pkg/deploy/util/util.go

```diff
@@ -228,6 +234,19 @@ func EncodeDeploymentConfig(config *deployapi.DeploymentConfig, codec runtime.Co
 	return string(bytes[:]), nil
 }
 
+func NewControllerRef(config *deployapi.DeploymentConfig) *metav1.OwnerReference {
+	blockOwnerDeletion := true
+	isController := true
+	return &metav1.OwnerReference{
+		APIVersion:         ControllerKind.Version,
```

Comment:
My guess is that it does not matter, as long as the referenced object exists. But I am not deep in the GC algorithm either.